interpreter/per-host/src/main/scala/io/buoyant/transformer/perHost/PortTransformer.scala

```diff
@@ -26,19 +26,27 @@ class PortTransformer(port: Int) extends DelegatingNameTreeTransformer {
         Addr.Bound(addrs.map(mapAddress), meta)
       case addr => addr
     }
-    Name.Bound(vaddr, bound.id, bound.path)
+    bound.id match {
+      case id: Path => Name.Bound(vaddr, prefix ++ id, bound.path)
+      case _ => Name.Bound(vaddr, bound.id, bound.path)
+    }
   }
 
   override protected def transform(tree: NameTree[Bound]): Activity[NameTree[Bound]] =
     Activity.value(tree.map(mapBound))
 
   override protected def transformDelegate(tree: DelegateTree[Bound]): Activity[DelegateTree[Bound]] =
     Activity.value(tree.flatMap { leaf =>
+      val bound = mapBound(leaf.value)
+      val path = bound.id match {
+        case id: Path => id
+        case _ => leaf.path
+      }
```

Comment:
I've factored this common code into the DelegatingNameTreeTransformer companion object and renamed variables for clarity.
Here we are extracting the transformed id from the transformed `Name.Bound`.  

`bound.id` should be equal to `prefix ++ orig.id`.